Constraints/include/TaperedCapsuleCollision.h

```diff
@@ -9,7 +9,9 @@
 namespace RAPTOR {
 
 #define COLLISION_THRESHOLD 1e-4
-#define NUM_FACTORS 2
+#ifndef NUM_FACTORS
```

Comment:
Nice!
Could you also manage to add this in `Armour` namespace please? Just to avoid potential conflicts.

Reply:
I wonder if it's possible to define a logic like
if NUM_FACTORS is defined in Armour namespace, then use this fixed size matrix.
otherwise, we should still use dynamical size matrix, since this class should be general.

Reply:
I think this should be a templated class with an equivalent to num_factors to define the size of the arrays. This gets the speed of having fixed sizes while allowing the class to be general